commands/new_function.go

```diff
@@ -157,21 +172,19 @@ func runNewFunction(cmd *cobra.Command, args []string) error {
 	var stackYaml string
 
 	if !appendMode {
-		stackYaml +=
-			`provider:
-  name: faas
-  gateway: ` + gateway + `
-
-functions:
-`
+		stackYaml += fmt.Sprintf(
+			headerYaml,
+			gateway,
+		)
 	}
 
-	stackYaml +=
-		`  ` + functionName + `:
-    lang: ` + language + `
-    handler: ./` + functionName + `
-    image: ` + imageName + `
-`
+	stackYaml += fmt.Sprintf(
```

Comment:
I'd asked for the Sprintf approach to be removed. The Sprintf is unreadable and hard to change and someone still has to scroll to find the definition. A Go template would be better because it's not sensitive to the ordering.

Example in practice with an in-line template: https://golang.org/pkg/text/template/#example_Template

I don't believe an external file is required at this stage.